src/main/java/org/gridsuite/actions/server/ContingencyListService.java

```diff
@@ -290,12 +290,14 @@ private boolean filterByVoltage(double equipmentNominalVoltage, double nominalVo
             .or(() -> getFiltersContingencyList(id).map(contingencyList -> toPowSyBlContingencyList(contingencyList, networkUuid)));
     }
 
-    ScriptContingencyList createScriptContingencyList(ScriptContingencyList script) {
+    ScriptContingencyList createScriptContingencyList(UUID id, ScriptContingencyList script) {
         Objects.requireNonNull(script.getName());
         if (LOGGER.isDebugEnabled()) {
             LOGGER.debug("Create script contingency list '{}'", sanitizeParam(script.getName()));
         }
-        return fromScriptContingencyListEntity(scriptContingencyListRepository.save(new ScriptContingencyListEntity(script)));
+        ScriptContingencyListEntity entity = new ScriptContingencyListEntity(script);
+        entity.setId(id == null ? UUID.randomUUID() : id);
```

Comment:
From my point of view this api should only be accessed through the directory-server so the ID should always be there

Reply:
See my remark above